Filter out "untranslated preview pictures" and RAWs from JapScan.

AmineI opened the pull request:
Changed the CSS selector to exclude chapter names  containing "Spoiler SPOILER" as described in
inorichi/tachiyomi#1925 

(would be great to somehow separate that feature request above and the issue described in it, to actually put that issue here instead.)

Still draft as I'm awaiting confirmation that these chapters are ALWAYS named "Spoiler SPOILER".

AmineI commented:
Found this (raws) while searching on the website. Might as well remove them, as they are replaced as soon as they are translated.

![image](https://user-images.githubusercontent.com/15179425/54921791-ed3b5f00-4f06-11e9-9c8b-78e4585e9280.png)

AmineI commented:
I now rely on the website's badges to detect if a chapter is a raw or a preview, so it should be good to go if the spoiler badge works the same as the "raw" or "vus" badge, but I'm yet to observe a spoiler badge.

I don't use Japscan, so if you (@pokke456) notice any new spoiler chapter on the website, please tell me so I can confirm that.

CarlosEsco commented:
@AmineI  good to merge?

AmineI commented:
Filtering out RAWs is fine and tested but I couldn’t test with "untranslated preview pictures" as there aren’t any on their website RN.
They seem to be consistent with how they use badges, so it should work properly, but it’s untested.

Worst case is that the preview pictures won’t be excluded. Might as well wait a few days until they upload one to be sure before we merge ?

De : Carlos<mailto:notifications@github.com>
Envoyé le :lundi 25 mars 2019 16:09
À : inorichi/tachiyomi-extensions<mailto:tachiyomi-extensions@noreply.github.com>
Cc : Amine A.<mailto:amine-98@live.fr>; Mention<mailto:mention@noreply.github.com>
Objet :Re: [inorichi/tachiyomi-extensions] Filter out "untranslated preview pictures" and RAWs from JapScan. (#972)


@AmineI<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAmineI&data=02%7C01%7C%7Ccde0c6d32cec49c6c2ab08d6b133d623%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636891233519491676&sdata=FH8z0VsnMBX9sP81xaS%2Be3%2BCsjk4bvQoN3Lx6Mxv0mM%3D&reserved=0> good to merge?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Finorichi%2Ftachiyomi-extensions%2Fpull%2F972%23issuecomment-476239469&data=02%7C01%7C%7Ccde0c6d32cec49c6c2ab08d6b133d623%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636891233519501675&sdata=%2FfVzd%2BnQrpqfJpNzJwGtTMuOSOz4Q0ZaumWBLnPBOWs%3D&reserved=0>, or mute the thread<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAOeeoREZlTPrX35YW7w0DpQtgft-kNdyks5vaOaWgaJpZM4cG0Mi&data=02%7C01%7C%7Ccde0c6d32cec49c6c2ab08d6b133d623%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636891233519511692&sdata=7TfekRfP0wD%2BXuAAca05mfKS26FZvRieUf1ILsIBAI4%3D&reserved=0>.

Peuuuur-Noel commented on src/fr/japscan/src/eu/kanade/tachiyomi/extension/fr/japscan/Japscan.kt:
```diff
@@ -159,14 +159,19 @@ class Japscan : ParsedHttpSource() {
 
-    override fun chapterListSelector() = "div#chapters_list > div.collapse > div.chapters_list"
+    override fun chapterListSelector() = "#chapters_list > div.collapse > div.chapters_list"+
+            ":not(:has(.badge:contains(SPOILER),.badge:contains(RAW)))"
```
You can add, in the exclude, `VUS` badge which correspond of the chapter in the Version US. Waiting for the French translation.
Like for this one: https://www.japscan.to/manga/the-promised-neverland/

AmineI replied:
That's why I asked what it was indeed, as we said it's better to exclude them - thanks :) .

pokke456 commented:
> I don't use Japscan, so if you (@pokke456) notice any new spoiler chapter on the website, please tell me so I can confirm that.

👍 
They should publish those spoilers on thursday, before the real chapters released. I'll let you know

AmineI commented:
They uploaded one of these preview "chapters" today, and it was used as expected so no changes needed about that. added another exclude filter while we're there. We're good to merge @CarlosEsco :)

Note : I deleted a message above which was an answer to CarlosEsco, due to some text added by my e-mail client that I wished private. My bad, here's the message anyway

>Filtering out RAWs is fine and tested but I couldn’t test with "untranslated preview pictures" as there aren’t any on their website RN.
They seem to be consistent with how they use badges, so it should work properly, but it’s untested.
Worst case is that the preview pictures won’t be excluded. Might as well wait a few days until they upload one to be sure before we merge ?
